internal/cache/file/downloader/job_test.go

```diff
@@ -280,3 +309,90 @@ func (jt *jobTest) Test_updateFileInfoCache_Fail() {
 	lookupResult := jt.cache.LookUp(fileInfoKeyName)
 	ExpectTrue(lookupResult == nil)
 }
+
+// Note: We can't test Test_downloadObjectAsync_MoreThanSequentialReadSize as
+// the fake storage bucket/server in the testing environment doesn't support
+// reading ranges (start and limit in NewReader call)
+func (jt *jobTest) Test_downloadObjectAsync_LessThanSequentialReadSize() {
+	// Create new object in bucket and create new job for it.
+	objectName := "path/in/gcs/foo.txt"
+	objectSize := 50 * MiB
+	objectContent := bytes.Repeat([]byte("t"), objectSize)
+	jt.initJobTest(objectName, objectContent, 100, uint64(2*objectSize))
+
+	// start download
+	jt.job.downloadObjectAsync()
+
+	// check job completed successfully
+	jobStatus := JobStatus{COMPLETED, nil, int64(objectSize)}
+	jt.job.mu.Lock()
+	defer jt.job.mu.Unlock()
+	ExpectTrue(reflect.DeepEqual(jobStatus, jt.job.status))
+	// verify file is downloaded
+	jt.verifyFile(objectContent)
+	// Verify fileInfoCache update
+	jt.verifyFileInfoEntry(uint64(objectSize))
+}
+
+func (jt *jobTest) Test_downloadObjectAsync_LessThanChunkSize() {
+	objectName := "path/in/gcs/foo.txt"
+	objectSize := 2 * MiB
```

Comment:
Same as above comment - we should make the test case realistic. objectSize == len(objectContent)?